src/main/java/de/mediathekview/mserver/crawler/srf/parser/SrfFilmJsonDeserializer.java

```diff
@@ -0,0 +1,351 @@
+        if (showObject.has(ATTRIBUTE_TITLE)) {
+          return showObject.get(ATTRIBUTE_TITLE).getAsString();
+        }
+      }
+    }
+
+    return "";
+  }
+
+  private EpisodeData parseEpisode(JsonObject aJsonObject) {
+    EpisodeData result = new EpisodeData();
+
+    if (aJsonObject.has(ELEMENT_EPISODE)) {
+      JsonElement episodeElement = aJsonObject.get(ELEMENT_EPISODE);
+
+      if (!episodeElement.isJsonNull()) {
+        JsonObject episodeObject = episodeElement.getAsJsonObject();
+
+        if (episodeObject.has(ATTRIBUTE_TITLE)) {
+          result.title = episodeObject.get(ATTRIBUTE_TITLE).getAsString();
+        }
+
+        if (episodeObject.has(ATTRIBUTE_PUBLISHED_DATE)) {
+          String date = episodeObject.get(ATTRIBUTE_PUBLISHED_DATE).getAsString();
+          result.publishDate = LocalDateTime.parse(date, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+        }
+      }
+    }
+
+    return result;
+  }
+
+  private ChapterListData parseChapterList(JsonObject aJsonObject) {
+    ChapterListData result = new ChapterListData();
+
+    if (aJsonObject.has(ELEMENT_CHAPTER_LIST)) {
+      JsonElement chapterListElement = aJsonObject.get(ELEMENT_CHAPTER_LIST);
+
+      if (!chapterListElement.isJsonNull()) {
+        JsonArray chapterListArray = chapterListElement.getAsJsonArray();
+
+        if (chapterListArray.size() == 1) {
+          JsonObject chapterListEntry = chapterListArray.get(0).getAsJsonObject();
+
+          if (chapterListEntry.has(ATTRIBUTE_ID)) {
+            result.id = chapterListEntry.get(ATTRIBUTE_ID).getAsString();
+          }
+
+          if (chapterListEntry.has(ATTRIBUTE_DURATION)) {
+            long duration = chapterListEntry.get(ATTRIBUTE_DURATION).getAsLong();
+            result.duration = Duration.of(duration, ChronoUnit.MILLIS);
+          }
+
+          if (chapterListEntry.has(ATTRIBUTE_DESCRIPTION)) {
+            result.description = chapterListEntry.get(ATTRIBUTE_DESCRIPTION).getAsString();
+          }
+
+          if (chapterListEntry.has(ELEMENT_RESOURCE_LIST)) {
+            result.videoUrl = parseResourceList(chapterListEntry.get(ELEMENT_RESOURCE_LIST));
+          }
+          
+          if (chapterListEntry.has(ELEMENT_SUBTITLE_LIST)) {
+            result.subtitleUrl = parseSubtitleList(chapterListEntry.get(ELEMENT_SUBTITLE_LIST));
+          }
+        }
+      }
+    }
+
+    return result;
+  }
+  
+  private static String parseSubtitleList(JsonElement aSubtitleListElement) {
+    if (!aSubtitleListElement.isJsonArray()) {
+      return "";
+    }
+
+    JsonArray subtitleArray = aSubtitleListElement.getAsJsonArray();
+    for (JsonElement arrayItemElement : subtitleArray) {
+      if (!arrayItemElement.isJsonNull()) {
+        JsonObject arrayItemObject = arrayItemElement.getAsJsonObject();
+
+        if (arrayItemObject.has(ATTRIBUTE_FORMAT) && arrayItemObject.has(ATTRIBUTE_URL)) {
+          if (arrayItemObject.get(ATTRIBUTE_FORMAT).getAsString().equals(SUBTITLE_FORMAT)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)